code/game/objects/structures/props/beam_prism.dm

```diff
@@ -0,0 +1,218 @@
+	plane = -10					//Layer over projectiles.
+
+	var/rotation_lock = 0		// Can you rotate the prism at all?
+	var/free_rotate = 1			// Does the prism rotate in any direction, or only in the eight standard compass directions?
+	var/external_control_lock = 0	// Does the prism only rotate from the controls of an external switch?
+	var/degrees_from_north = 0	// How far is it rotated clockwise?
+	var/compass_directions = list("North" = 0, "South" = 180, "East" = 90, "West" = 270, "Northwest" = 315, "Northeast" = 45, "Southeast" = 135, "Southwest" = 225)
+	var/interaction_sound = 'sound/mecha/mechmove04.ogg'
+
+	var/redirect_type = /obj/item/projectile/beam
+
+	var/dialID = null
+	var/obj/structure/prop/prismcontrol/remote_dial = null
+
+	interaction_message = "<span class='notice'>The prismatic turret seems to be able to rotate.</span>"
+
+/obj/structure/prop/prism/initialize()
+	if(degrees_from_north)
+		animate(src, transform = turn(NORTH, degrees_from_north), time = 3)
+
+/obj/structure/prop/prism/Destroy()
+	if(remote_dial)
+		remote_dial.my_turrets -= src
+		remote_dial = null
+	..()
+
+/obj/structure/prop/prism/proc/reset_rotation()
+	var/degrees_to_rotate = -1 * degrees_from_north
+	animate(src, transform = turn(src.transform, degrees_to_rotate), time = 2)
+
+/obj/structure/prop/prism/attack_hand(mob/living/user)
+	..()
+
+	if(rotation_lock)
+		to_chat(user, "<span class='warning'>\The [src] is locked at its current bearing.</span>")
+		return
+	if(external_control_lock)
+		to_chat(user, "<span class='warning'>\The [src]'s motors resist your efforts to rotate it. You may need to find some form of controller.</span>")
+		return
+
+	var/confirm = input("Do you want to try to rotate \the [src]?", "[name]") in list("Yes", "No")
+	if(confirm == "No")
+		visible_message(\
+			"<span class='notice'>[user.name] decides not to try turning \the [src].</span>",\
+			"<span class='notice'>You decide not to try turning \the [src].</span>")
+		return
+
+	var/new_bearing
+	if(free_rotate)
+		new_bearing = input("What bearing do you want to rotate \the [src] to?", "[name]") as num
+		new_bearing = round(new_bearing)
+		if(new_bearing <= -1 || new_bearing > 360)
+			to_chat(user, "<span class='warning'>Rotating \the [src] [new_bearing] degrees would be a waste of time.</span>")
+			return
+	else
+		var/choice = input("What point do you want to set \the [src] to?", "[name]") as null|anything in compass_directions
+		new_bearing = round(compass_directions[choice])
+
+	var/rotate_degrees = new_bearing - degrees_from_north
+
+	if(new_bearing == 360) // Weird artifact.
+		new_bearing = 0
+	degrees_from_north = new_bearing
+
+	var/two_stage = 0
+	if(rotate_degrees == 180 || rotate_degrees == -180)
+		two_stage = 1
+		var/multiplier = pick(-1, 1)
+		rotate_degrees = multiplier * (rotate_degrees / 2)
+
+	playsound(src, interaction_sound, 50, 1)
+	if(two_stage)
+		animate(src, transform = turn(src.transform, rotate_degrees), time = 3)
+		sleep(3)
```

Comment:
I think putting this inside a `spawn(3)` will have the same effect, and be desirable because spawn's nonblocking

Reply:
This is the same method that crawling uses, which is what I based the entirety of the animation code on, but I can change it if necessary